src/test/java/org/opentripplanner/gtfs/mapping/FareLegRuleMapperTest.java

```diff
@@ -53,7 +53,7 @@ Stream<DynamicTest> mapDistance() {
             fp.setFareProductId(new AgencyAndId("1", "1"));
 
             var obaRule = new FareLegRule();
-            obaRule.setFareProduct(fp);
+            obaRule.setFareProductId(fp.getFareProductId());
```

Comment:
It's a bit of work but can you please add a test case for multiple fare products in a rule? You will have to refactor this test a bit.

Reply:
Done, how's that? I also wanted to also test the null/empty fare products checks that you recommended be added in FareLegRule but couldn't think of a good way, so I just made sure that the rule wouldn't be mapped properly if there was no product ID provided (I was also wondering if it would be possible to check the `issueStore` in the `FareProductMapper` but couldn't immediately see a good way to do that either).

Reply:
You can test the issue but injecting a `DefaultDataImportIssueStore` (not DataImportIssueStore.NOOP) and then asserting that it has the correct issues in it. However, this is really not required for me to get this merged.